File: src/Java/myservicebus/src/main/java/com/myservicebus/serialization/MassTransitHeaderConvention.java
package com.myservicebus.serialization;

import com.myservicebus.MessageHeaders;

public final class MassTransitHeaderConvention implements MessageHeaderConvention {
    public static final MassTransitHeaderConvention INSTANCE = new MassTransitHeaderConvention();

    private MassTransitHeaderConvention() {
    }

    @Override
    public String getContentTypeHeader() {
        return "content_type";
    }

    @Override
    public String getFaultAddressHeader() {
        return MessageHeaders.FAULT_ADDRESS;
    }

    @Override
    public boolean isHostHeader(String headerName) {
        return headerName.startsWith("MT-Host-");
    }
}

File: src/MyServiceBus/Serialization/MassTransitHeaderConvention.cs
using System;

namespace MyServiceBus.Serialization;

public sealed class MassTransitHeaderConvention : IMessageHeaderConvention
{
    public static MassTransitHeaderConvention Instance { get; } = new();

    private MassTransitHeaderConvention()
    {
    }

    public string ContentTypeHeader => "content_type";

    public string FaultAddressHeader => MessageHeaders.FaultAddress;

    public bool IsHostHeader(string headerName)
        => headerName.StartsWith("MT-Host-", StringComparison.Ordinal);
}
